python/ray/util/sgd/torch/examples/tune_example.py

```diff
@@ -56,6 +64,39 @@ def tune_example(operator_cls, num_workers=1, use_gpu=False):
 # __end_torch_tune_example__
 
 
+# __torch_tune_manual_lr_example__
+def tune_example_manual(operator_cls, num_workers=1, use_gpu=False):
+    def tune_func(trainer, iter):
+        """Define a custom training loop for tune.
+         This is needed because we want to manually update our scheduler.
+         """
+        train_stats = trainer.train(profile=True)
+        validation_stats = trainer.validate(profile=True)
+        # Manually update our scheduler with the given metric.
+        trainer.update_scheduler(metric=validation_stats["val_loss"])
+        all_stats = merge_dicts(train_stats, validation_stats)
+        return all_stats
+
+    TorchTrainable = TorchTrainer.as_trainable(
+        tune_func=tune_func,
```

Comment:
This is a callback right? tune_func is an odd function name. 

What about 'on_train_step', 'on_step', 'callback'?

Reply:
I'd vote for `on_step` here. Then we could (potentially) allow other methods to be overwritten, too. Also, the example code includes validation, so I'd vote agains including `train` in the function name.

Reply:
Yep makes sense- `tune_func` is not a great name. I changed it to just `step` for now. I'm not sure if `on_step` is the best name because I feel that it implies that it's more of a side effect function, but in this case, it's actually supposed to encapsulate the main training loop. Obviously, this is open to further discussion.